packages/block-editor/src/utils/theme.js

```diff
@@ -3,6 +3,16 @@
  */
 import { SETTINGS_DEFAULTS } from '../store/defaults';
 
+export function __experimentalGetHighestPriorityPreset(
```

Comment:
Can this be absorbed by the `useSetting` hook? I'd imagine this would work this way:

- `useSetting( 'color.palette' )`: returns the consolidated / merged palette.
- `useSetting( 'color.palette.core' )`: returns the core colors.
- `useSetting( 'color.palette.theme' )`: returns the theme colors.
- `useSetting( 'color.palette.user' )`: returs the user colors.

Similarly for the other presets.

Reply:
I suggested this option in https://github.com/WordPress/gutenberg/pull/32358#issuecomment-852250830.
@youknowriad seemed to prefer to avoid this logic inside useSetting https://github.com/WordPress/gutenberg/pull/32358#issuecomment-852857749.

Reply:
I'm hesitant :) it feels like we shouldn't allow `color.palette` entirely

Reply:
Is your concern it won't serve any purpose with the upcoming UI changes? Mine is similar: introducing a temporary `__experimentalGetHighestProiorityPreset` that will be deprecated, with the additional burden it poses in the consumers (they need to know the inner details of how this works).

Thinking out loud: the `settings.colors` should still have the proper values for the UI https://github.com/WordPress/gutenberg/pull/32358#discussion_r644234638 so I wonder if a way to avoid introducing this temporary utility would be to access `settings.colors` directly (same for gradients and font sizes)?

Reply:
> Is your concern it won't serve any purpose with the upcoming UI changes?

Yes, it's not entirely clear how the global `color.palette` is going to be used. `__experimentalGetHighestProiorityPreset` is experimental so it's fine but remains the question about `color.palette`

Reply:
I think you should trust your instinct here, I personally don't have the right answer :)